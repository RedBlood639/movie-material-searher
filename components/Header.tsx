import { useReactiveVar } from '@apollo/client'
import styled from '@emotion/styled'
import { AppBar, IconButton, Toolbar } from '@material-ui/core'
import {
  Brightness4 as DarkIcon,
  Brightness7 as LightIcon,
} from '@material-ui/icons'
import Link from 'next/link'

import { prefersDarkModeVar } from '@/apollo/client'

import Favorites from './Favorites'
import Logo from './Logo'
import SearchBar from './SearchBar'

const StyledToolbar = styled(Toolbar)`
  padding: ${({ theme }) => theme.spacing(0, 1)};
  ${({ theme }) => theme.breakpoints.up('sm')} {
    padding: ${({ theme }) => theme.spacing(0, 2)};
  }
`
const StyledIconButton = styled(IconButton)`
  margin-left: auto;
`

export default function Header() {
  const prefersDarkMode = useReactiveVar(prefersDarkModeVar)

  const toggleTheme = () => {
    const toggle = !prefersDarkMode
    prefersDarkModeVar(toggle)
    localStorage.setItem('darkMode', JSON.stringify(toggle))
  }

  return (
    <AppBar color="inherit" position="fixed">
      <StyledToolbar disableGutters>
        <Link href="/">
          <IconButton aria-label="go to home page">
            <Logo prefersDarkMode={prefersDarkMode} />
          </IconButton>
        </Link>
        <SearchBar />
        <StyledIconButton aria-label="theme switch" onClick={toggleTheme}>
          {prefersDarkMode ? <LightIcon /> : <DarkIcon />}
        </StyledIconButton>
        <Favorites />
      </StyledToolbar>
    </AppBar>
  )
}
