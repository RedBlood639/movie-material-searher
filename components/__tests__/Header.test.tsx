import { fireEvent, renderApollo, screen } from '@/lib/setupTests'

import Header from '../Header'

describe('header', () => {
  it('should toggle theme and save to local storage', () => {
    const { baseElement } = renderApollo(<Header />)

    const button = screen.getByLabelText(/theme switch/i)

    fireEvent.click(button)

    expect(baseElement).toMatchSnapshot()
  })
})
