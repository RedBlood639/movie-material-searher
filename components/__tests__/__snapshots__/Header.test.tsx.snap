// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should toggle theme and save to local storage 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorInherit MuiAppBar-positionFixed mui-fixed css-1vlwt05-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular e1aqnoh81 css-blhzm8-MuiToolbar-root-StyledToolbar"
      >
        <button
          aria-label="go to home page"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1uaxfgw-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label css-4jkopv-MuiIconButton-label"
          >
            <svg
              height="45"
              viewBox="0 0 65 65"
              width="45"
            >
              <g
                fill="#fff"
              >
                <path
                  d="m31 5c14.888 0 27 12.112 27 27s-12.112 27-27 27-27-12.112-27-27 12.112-27 27-27m0-3c-16.567 0-30 13.432-30 30s13.433 30 30 30 30-13.432 30-30-13.433-30-30-30z"
                />
                <path
                  d="m43.987 29.745-15.974-10.513c-2.757-1.814-5.013-.597-5.013 2.703v22.125c0 3.3 2.259 4.521 5.021 2.714l15.959-10.444c2.761-1.809 2.764-4.771.007-6.585zm-2.129 4.379-14.185 9.283c-.92.603-1.673.196-1.673-.904v-19.012c0-1.1.752-1.505 1.671-.9l14.19 9.338c.919.604.918 1.592-.003 2.195z"
                />
              </g>
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root ep0uteb4 css-78p48u-MuiAutocomplete-root-StyledAutocomplete"
          role="combobox"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium ep0uteb2 css-1cntsc7-MuiSvgIcon-root-StyledSearchIcon"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <div
            class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root ep0uteb3 css-1oyzkgj-MuiFormControl-root-StyledTextField"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-4a33cb-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused MuiAutocomplete-input MuiAutocomplete-inputFocused css-b55ri7-MuiInputBase-input-MuiOutlinedInput-input"
                id="autocomplete"
                placeholder="Search..."
                spellcheck="false"
                type="text"
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="css-1br1jja"
                  style="width: 0.01px;"
                >
                  <span
                    class="notranslate"
                  >
                    â€‹
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          aria-label="theme switch"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium e1aqnoh80 css-17n4xwh-MuiButtonBase-root-MuiIconButton-root-StyledIconButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label css-4jkopv-MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="Brightness7Icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="open favorites"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-1uaxfgw-MuiButtonBase-root-MuiIconButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiIconButton-label css-4jkopv-MuiIconButton-label"
          >
            <span
              class="MuiBadge-root css-1c32n2y-MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FavoriteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="MuiBadge-colorSecondary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRightRectangular MuiBadge-invisible css-13j2lht-MuiBadge-badge"
              >
                0
              </span>
            </span>
          </span>
        </button>
      </div>
    </header>
  </div>
</body>
`;
